tutorials/users/ide/version-control.md

```diff
@@ -1 +1,607 @@
+
+Fill out the form and input a branch name, and click which branch to base the new branch off of.  After that click `Create Branch`.
+
+It is good practice to not use spaces or special characters in the branch name other than dashes.
+
+
+![New Branch Form](images/new-branch-dialog.png)
+
+Git will automatically switch you into the new branch.  Your newly created branch will be shown under `Current Branch` and be in the list of branches.  
+
+![Create Branch Successful](images/create-branch-successful.png)
+
+
+### Switching Branches
+
+```{note}
+You can only switch branches when you have no conflicts in files between the branches.
+```
+
+To switch branches simply click another branch listed under the `Branches` section.
+
+```{note}
+You can only switch to another local branch.  Branches with the `origin` prefix are remote branches and cannot be switched into.
+``` 
+
+
+## Using Git with the Terminal
+
+In this section are some useful Git commands.  For more in-depth information take a look at the 
+[official Git documentation](https://git-scm.com/doc).
+
+### Set Up Your Git Identifier
+
+When you set up your Git Identifier your username and email will be shown with each commit you make.
+If you are working in a team this will show who made a certain commit.
+
+```
+git config --global user.name "Your Name"
+git config --global user.email "email@email.com"
+```
+
+### Staging a File
+
+A file can be staged whether you are in the same directory as the file or not. 
+
+To stage a file when you are in the same directory as the file, you use the command with only the filename.
+
+```
+git add <filename>
+```
+
+To stage a file when you are in a different directory as the file, you use the command with the folder path to the file.
+
+```
+git add /folder/path/to/file/<filename>
+```
+
+### View Status
+
+If you have multiple files staged and ready to commit you can see the list by using `git status`.
+
+```
+git status
+```
+
+
+### Making a Commit
+
+Add a brief descriptive message about your commit.
+
+```
+git commit -m "your brief commit message goes here"
+```
+
+### Pushing a Commit
+
+After your commit is complete, push it to the remote repository.
+
+```
+git push
+```
+
+### Pushing a Commit to an Existing Remote Repository
+If you want to push to a specific remote repository specify the name of the repository with the `-u` flag.
+`-u` is the same as using the full command `-set-upstream`.
+
+```
+git push -u remote-repository-name
+```
+
+You can also specify a specific branch to push to.
+
+```
+git push -u remote-repository-name remote-branch-name
+```
+
+### View Commit History
+
+To see all the commits made in a repository use the `log` command.
+
+```
+git log
+```
+
+### View the Changes Made Between Versions
+
+To view a comparison of the changes between versions of the same file run the `diff` command.
+
+```
+git diff
+```
+
+### Undo Changes
+
+1. Undo changes in a single file, but keep it staged:
```

Comment:
I think you need to provide some explanation what you mean by "commit tag" in these examples. I don't think you've introduced that concept yet

Reply:
Added explanation and example of a commit tag under View Commit History section